homeassistant/components/lock/homematic.py

```diff
@@ -0,0 +1,59 @@
+"""
+Support for HomeMatic switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.homematic/
+"""
+import logging
+from homeassistant.components.lock import LockDevice
+from homeassistant.components.homematic import HMDevice, ATTR_DISCOVER_DEVICES
+from homeassistant.const import (STATE_UNKNOWN, STATE_LOCKED, STATE_UNLOCKED)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['homematic']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the HomeMatic switch platform."""
+    if discovery_info is None:
+        return
+
+    devices = []
+    for conf in discovery_info[ATTR_DISCOVER_DEVICES]:
+        new_device = HMLock(conf)
+        devices.append(new_device)
+
+    add_devices(devices)
+
+
+class HMLock(HMDevice, LockDevice):
+    """Representation of a HomeMatic lock."""
+
+    def _init_data_struct(self):
+        """Generate the data dictionary (self._data) from metadata."""
+        self._state = "STATE"
+        self._data.update({self._state: STATE_UNKNOWN})
+
+    @property
+    def is_locked(self):
+        """Return true if device is on."""
+        try:
+            return self._hm_get_state() == 0
+        except TypeError:
+            return False
+
+    def lock(self, **kwargs):
+        """Lock the device."""
+        self._hmdevice.lock(self._channel)
+        self._state = STATE_LOCKED
+
+    def unlock(self, **kwargs):
+        """Unlock the device."""
+        self._hmdevice.unlock(self._channel)
+        self._state = STATE_UNLOCKED
+
+    def update(self):
+        """Update state by device callback."""
+        self._state = (STATE_LOCKED if self.is_locked()
+                       else STATE_UNLOCKED)
```

Comment:
I'm not a `Lock` component expert (nor a Homematic user), but I think this call is unnecessary because `LockDevice` already handles this for you:

https://github.com/home-assistant/home-assistant/blob/80d2c76e85afbf2da3786492bfa2ab3977bd601a/homeassistant/components/lock/__init__.py#L171-L177

Reply:
Oh ... sure .. ;-)